lib/PuppeteerSharp.Tests/Issues/Issue0616.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace PuppeteerSharp.Tests.Issues
+{
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class Issue0616 : PuppeteerPageBaseTest
+    {
+        public Issue0616(ITestOutputHelper output) : base(output)
+        {
+        }
+
+        [Fact]
+        public async Task ShouldBeAbleToChangeToPost()
+        {
+            await Page.SetRequestInterceptionAsync(true);
+            Page.Request += async (sender, e) =>
+            {
+                var payload = new Payload()
+                {
+                    Method = HttpMethod.Post,
+                    PostData = "foo"
+                };
+                await e.Request.ContinueAsync(payload);
+            };
+
+            Server.SetRoute("/grid.html", async (context) =>
+            {
+                using (var reader = new StreamReader(context.Request.Body, Encoding.UTF8))
+                {
+                    var request = await reader.ReadToEndAsync();
+                    await context.Response.WriteAsync(request);
+                }
+            });
+            var response = await Page.GoToAsync(TestConstants.ServerUrl + "/grid.html");
+
+            Assert.Equal(TestConstants.ServerUrl + "/grid.html", response.Url);
```

Comment:
I think you should test the response method too, it should be `POST` not `GET`

Reply:
I checked that on Puppeteer. The `Network.setRequestInterception` message doesn't override the original request values, so it returns a `GET`
<img width="513" alt="screen shot 2018-09-14 at 08 36 17" src="https://user-images.githubusercontent.com/2198466/45548096-76a40300-b7f9-11e8-8155-1fab58286e7d.png">